.github/workflows/build-and-publish.yml

```diff
@@ -11,106 +11,53 @@ on:
 ##  WORKFLOW JOBS
 ####################################################################################################
 jobs:
-  # check_diff job pulls the code and checks that updates to at least one csproj file have been made.
-  check_diff:
-    name: Fetch code and check diff
-    # Using the latest Windows GitHub-hosted runner.
-    runs-on: windows-latest
+  # set_variables sets variables needed for the workflow
+  set_variables:
+    name: Set variables
+    # Using the latest Ubuntu GitHub-hosted runner.
+    runs-on: ubuntu-latest
     # Creates output variables to be used in steps or other jobs.
     outputs: 
-      diff: ${{ steps.check-diff.outputs.diff }}
-      has_changes: ${{ steps.check-diff.outputs.has_changes }}
-  
-    steps:
-      # Using Github checkout action to retrieve the codebase with commit history.
-      - name: Fetch code with commit history
-        uses: actions/checkout@v3
-        with: 
-          fetch-depth: 0
+      package_name: ${{ steps.variables.outputs.package_name }}
+      package_version: ${{ steps.variables.outputs.package_version }}
 
-      # Using a powershell script to find which projects, if any, have been updated.
-      - name: Find updated project(s)
-        id: check-diff
-        shell: pwsh
+    steps:
+      # Set various variables needed in other jobs.
+      - name: Set variables
+        id: variables
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          # Get the diff between main to the previous commit to main.
-          $diff = git diff --name-only origin/main^...origin/main
-          echo "file(s) updated - $diff"  
+          # Gets release tag. 
+          releaseTag=$(gh api -H "Accept: application/vnd.github+json" /repos/${GITHUB_REPOSITORY}/releases | jq -r 'first | .tag_name')
+          echo "release tag: $releaseTag"
 
-          # Filter to only changed csproj files, excluding tests and examples.
-          $projChanged = $diff | Where-Object { $_ -Match '.csproj$' }
-          $projChanged = $projChanged | Where-Object { $_ -NotMatch '.Tests.csproj$' }
-          $projChanged = $projChanged | Where-Object { $_ -NotMatch 'Example.' }
-          $projChanged = $projChanged | ForEach-Object -Process { $_.Remove(0, $_.IndexOf('/') + 1).Replace(".csproj","") }
-          echo "project(s) updated - $projChanged"
-          
-          # Set hasChanges to true if there are any changed projects after filtering.
-          $hasChanges = $projChanged.Length -gt 0
-          # Set singleProject to true if projChanged is not already an array of projects.
-          $singleProject = $projChanged -isnot [array]
+          # Gets package name from the release tag.
+          packageName=${releaseTag%%?[[:digit:]]*}
```

Comment:
Does this mean that we are manually creating releases via github and they have to be in a specific format?  And looks like we'd have to do one at a time right?

Reply:
Well I forgot that we were manually creating the release - https://git.rockfin.com/Rocket-Libraries/Process-Documentation/blob/main/deployments/rocklib-github-actions-release-proces.md so that process is the same which is fine.

Reply:
But I still think this would be one at a time so we might want to add it to the process documentation.